### PR TITLE
chore(operator): Upgrade operator-sdk 1.37.0 and kind 0.23.0

### PR DESCRIPTION
**Pull request by periklis**
**What this PR does / why we need it**:
Follow up on #14416 to upgrade the used `operator-sdk` and `kind` for local development.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [x] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: operator/config/overlays/community/manager_auth_proxy_patch.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ spec:
     spec:
       containers:
       - name: kube-rbac-proxy
-        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.15.0
+        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.16.0
```
Comment (JoaoBraveCoding) [inline operator/config/overlays/community/manager_auth_proxy_patch.yaml:10]: Why was this also updated? And why v0.16.0? It seems like v0.18.0 is the one aligned with k8s v1.30

Comment (periklis) [reply operator/config/overlays/community/manager_auth_proxy_patch.yaml:10]: @JoaoBraveCoding Following the migration steps from `v1.33` up to `v1.37` I was instructed to update to `v0.16` here: https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.36.0/

You say `kube-rbac-proxy@v0.18.0` is using the `k8s@v1.30` client-go/api then we can update to this one too. I need to double check what version we issue on OCP v4.17 though.

Comment (JoaoBraveCoding) [reply operator/config/overlays/community/manager_auth_proxy_patch.yaml]: I see! Yes according to the release notes that seems to be the case https://github.com/brancz/kube-rbac-proxy/releases

**Review by JoaoBraveCoding — APPROVED** [review]
lgtm

**Pull request merged**